### ISSUE TITLE
Consider supporting `-E`/`--preserve-env`

### ISSUE THREAD

**Issue by joshtriplett** [microsoft/sudo#21]
`sudo` for Linux supports a `-E`/`--preserve-env` option, which preserves environment variables from the current session when running the command. (By default, `sudo` should reset the environment variables to be those of the target user.)

`--preserve-env` additionally accepts an optional list of environment variables, to only preserve those variables rather than the entire environment.

This is sometimes useful when trying to run a command "logically" as your own user but which requires elevated permissions.

**Comment by gerardog** [comment]
How is that different with the current -E option?

![image](https://github.com/microsoft/sudo/assets/3901474/04a208c0-ac51-4b0c-a405-d7fcbe0a9b04)

**Comment by joshtriplett** [comment]
@gerardog That wasn't mentioned in the documentation at https://learn.microsoft.com/en-us/windows/sudo/ , so I didn't know it was supported. I've corrected this issue to just request `--preserve-env` and the optional list of environment variables.

**Comment by zadjii-msft** [comment]
Yep it was a total brainfart to not name it the same thing. Fortunately, this is super easy for us to fix, thanks!

**Comment by zadjii-msft** [comment]
Merged to OS in !10282477, which was OS.2020 commit 91ba4484. Thanks!

**Comment by joshtriplett** [comment]
@zadjii-msft Thank you! Does that include `--preserve-env=VAR1,VAR2`, in addition to renaming the long option?
